src/Servers/Kestrel/Core/src/KestrelServerOptions.cs

```diff
@@ -65,6 +71,24 @@ public class KestrelServerOptions
         /// </remarks>
         public bool DisableStringReuse { get; set; } = false;
 
+        /// <summary>
+        /// Controls whether to return the AltSvcHeader from on an HTTP/2 or lower response for HTTP/3
+        /// </summary>
+        /// <remarks>
+        /// Defaults to false.
+        /// </remarks>
+        public bool EnableAltSvc { get; set; } = false;
+
+        /// <summary>
+        /// Gets or sets a callback that returns the <see cref="Encoding"/> to decode the value for the specified request header name,
+        /// or <see langword="null"/> to use the default <see cref="UTF8Encoding"/>.
+        /// </summary>
+        public Func<string, Encoding> RequestHeaderEncodingSelector
```

Comment:
If you'd like to do additional legwork, you could turn on `nullable annotations` in the project, turn off nullable warnings, and turn this in to `Func<string, Encoding?>`.

Reply:
Talked offline. Even though this will be `Func<string, Encoding?>` once annotated, doing this now would require us it do it for the entire project, so we'll leave that for later. I tried doing this initially, but had to [undo the annotations](https://github.com/dotnet/aspnetcore/pull/23233/commits/4cf663aec36fbaedc9ff2810ba0696639b002bc2).